mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -418,25 +416,14 @@ void AbstractControllerExecution::run()
         condition_.notify_all();
       }
 
-      boost::chrono::thread_clock::time_point end_time = boost::chrono::thread_clock::now();
-      boost::chrono::microseconds execution_duration =
-          boost::chrono::duration_cast<boost::chrono::microseconds>(end_time - loop_start_time);
-      configuration_mutex_.lock();
-      boost::chrono::microseconds sleep_time = calling_duration_ - execution_duration;
-      configuration_mutex_.unlock();
       if (moving_ && ros::ok())
       {
-        if (sleep_time > boost::chrono::microseconds(0))
-        {
-          // interruption point
-          boost::this_thread::sleep_for(sleep_time);
-        }
-        else
+        if (!loop_rate_.sleep())
         {
           // provide an interruption point also with 0 or negative sleep_time
           boost::this_thread::interruption_point();
```

Comment:
If loop_rate_.sleep() doesn't honor interruptions, we should move this out of the if